jwst/extract_1d/extract.py

```diff
@@ -2586,6 +2588,17 @@ def run_extract1d(
         Switch to select whether or not to apply a 1d residual fringe correction
         for MIRI MRS IFU spectra.  Default is False.
 
+    ifu_set_point_source : bool
```

Comment:
Instead of having a param that just hardwires the srctype to POINT (even if it's already set to POINT upstream), would it be useful and more flexible to just have the param allow the user to choose which srctype they want, i.e. either POINT or EXTENDED? Hence instead of a boolean that just hardwires to POINT, make it a string with two allowed options of POINT and EXTENDED.

Reply:
I thought about that but I can not think when anyone would want the EXTENDED code and extracting the entire field of view. But I can put it in

Reply:
You could make it a generic param for setting the srctype, that would then get passed to and used by all modes.

Reply:
Is that not what the [srctype step parameter](https://jwst-pipeline.readthedocs.io/en/latest/jwst/srctype/arguments.html) is already? If a user sets this, will it not propagate to all future steps?